### PR TITLE
TF-14957 Add Team Tokens Index API

### PR DESCRIPTION
**Pull request by netramali**
### What and Why

Note: Do not merge until feature is released to customers.

This PR adds a new api documentation:  /organizations/:name/team-tokens
The API does the following:
- Users can get a list of all team authentication tokens in an organization, by default it orders by the expiration date
- The user can sort by team name, created by, expired at and last used
- The user can search the authentication token by the team name

[Jira](https://hashicorp.atlassian.net/browse/TF-13530)
### Screenshots

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. HCP Terraform ).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [ ] The Vercel website preview successfully deployed.

#### Reviews
- [ ] I or someone else reviewed the content for technical accuracy.
- [ ] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 1]
```diff
@@ -9,6 +9,11 @@ description: >-
 
 Keep track of changes to the API for HCP Terraform and Terraform Enterprise.
 
+## 2024-06-20
+* Add documentation for new [team tokens](/terraform/cloud-docs/api-docs/organization-team-tokens) API that lists team tokens in an organization.
+* Add reference to organization team token api on teams, index and navigation page.
+* Add a `sort` parameter to the [Team Tokens list API](/terraform/cloud-docs/api-docs/organization-team-tokens#query-parameters) to allow sorting tokens by team name, created by, expired at and last used.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/changelog.mdx:15]: I also don't think we need this bullet point because it's covered by the first bullet point. 
```suggestion
```

File: website/data/cloud-docs-nav-data.json [hunk 2]
```diff
@@ -134,6 +134,7 @@
       { "title": "Subscriptions", "path": "api-docs/subscriptions" },
       { "title": "Team Membership", "path": "api-docs/team-members" },
       { "title": "Team Tokens", "path": "api-docs/team-tokens" },
+      { "title": "Organization Team Tokens", "path": "api-docs/organization-team-tokens" },
```
Comment (rkoron007) [inline website/data/cloud-docs-nav-data.json:137]: Instead of creating a separate page for this, I do think we should either add this to the [Team token page](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/team-tokens) or honestly I think the [main organization API page](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/organizations) would be a good fit too. Since it's only one endpoint I think it could fit well on either page.

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 3]
```diff
@@ -9,6 +9,11 @@ description: >-
 
 Keep track of changes to the API for HCP Terraform and Terraform Enterprise.
 
+## 2024-06-20
+* Add documentation for new [team tokens](/terraform/cloud-docs/api-docs/organization-team-tokens) API that lists team tokens in an organization.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/changelog.mdx:13]:
```suggestion
* Add documentation for new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/organization-team-tokens).
```

File: website/docs/cloud-docs/api-docs/index.mdx [hunk 4]
```diff
@@ -150,6 +150,8 @@ The following entitlements are available:
 
 [team tokens]: /terraform/cloud-docs/api-docs/team-tokens
 
+[organization team tokens]: /terraform/cloud-docs/api-docs/organization-team-tokens
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/index.mdx:153]: Did you mean to link to this somewhere in this doc?

File: website/docs/cloud-docs/users-teams-organizations/teams.mdx [hunk 5]
```diff
@@ -30,6 +30,7 @@ In addition to the HCP Terraform UI, you can use the following methods to manage
 - [Teams API](/terraform/cloud-docs/api-docs/teams) to list, create, update, and delete teams
 - [Team Members API](/terraform/cloud-docs/api-docs/team-members) to add and delete users from teams
 - [Team Tokens API](/terraform/cloud-docs/api-docs/team-tokens) to generate and delete tokens
+- [Organization Team Tokens API](/terraform/cloud-docs/api-docs/organization-team-tokens) to list tokens
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/teams.mdx:33]: I don't think we need to mention this endpoint here because it doesn't actively _manage_ a team (though it'd definitely help you understand it better! but that's not what we are listing here)
```suggestion
```

File: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+[JSON API document]: /terraform/cloud-docs/api-docs#json-api-documents
+
+[JSON API error object]: https://jsonapi.org/format/#error-objects
+
+# Team Tokens API
+
+## List Team Tokens
+
+List team tokens in an organization.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/organization-team-tokens.mdx:43]:
```suggestion
Lists the [team tokens](/terraform/cloud-docs/users-teams-organizations/teams#api-tokens) in an organization.
```

File: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+`GET organizations/:organization_name/team-tokens`
+
+| Parameter            | Description                                              |
+|----------------------|----------------------------------------------------------|
+| `:organization_name` | The name of the organization the team tokens belong to |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/organization-team-tokens.mdx:49]:
```suggestion
| `:organization_name` | The name of the organization whose team tokens you want to list. |
```

File: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+The objects returned by this endpoint only contain metadata, and do not include the secret text of any authentication tokens. A token is only shown upon creation, and cannot be recovered later.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/organization-team-tokens.mdx:51]:
```suggestion
This endpoint returns object metadata and does not include secret authentication details of any token. You can only view a token when you create it and cannot recover it later.
```

File: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+By default, the tokens returned are ordered by ascending expiration date.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/organization-team-tokens.mdx:52]:
```suggestion
By default, this endpoint returns tokens by ascending expiration date.
```

File: website/docs/cloud-docs/api-docs/organization-team-tokens.mdx [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+
+| Status  | Response                                                | Reason                                                                                |
+| ------- | ------------------------------------------------------- | ------------------------------------------------------------------------------------- |
+| [200][] | [JSON API document][] (`type: "team-tokens"`)           | The request was successful                                                            |
+| [200][] | Empty [JSON API document][]                             | Organization has no team authentication tokens                                        |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/organization-team-tokens.mdx:57]:
```suggestion
| [200][] | Empty [JSON API document][]                             | The specified organization has no team tokens                                        |
```

**Review by rkoron007 — COMMENTED** [review]
First round of review! As always, please feel free to push back on anything that doesn’t feel good to you! ✨

File: website/docs/cloud-docs/api-docs/team-tokens.mdx
Comment (netramali) [inline website/docs/cloud-docs/api-docs/team-tokens.mdx:1]: @rkoron007 I wasn't sure what exactly I should include in the description.

Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/team-tokens.mdx:1]: Oh dear, it looks like this page doesn't have a description either! According to our style guide, we never "stack headers" by having one header following another without a sentence in between.
Screenshot 2024-06-28 at 12 45 22 PM


 Can you do me a favor and add a sentence or two on line 40 of this file below the `Team Token API` h1?  Just a quick intro about the end points on this page and what a team token does would be fantastic!

**Review by rkoron007 — COMMENTED** [review]
Left a few small comments, but this is looking great!

File: website/docs/cloud-docs/api-docs/team-tokens.mdx
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/team-tokens.mdx:1]: Oh, dear it looks like this page doesn't have a descriptive intro sentence! According to our style guide, we never want to stack headers, and always want at least a sentence or two between each heading.
Screenshot 2024-06-28 at 12 45 22 PM

Can you do me a favor and add an intro on line 40 of this file below the `Team Token API` h1? An introduction to what team tokens are, maybe [link to the teams page](https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/teams) and briefly describe each endpoint on the page.

Comment (netramali) [reply website/docs/cloud-docs/api-docs/team-tokens.mdx:1]: done!

File: website/docs/cloud-docs/api-docs/team-tokens.mdx [hunk 11]
```diff
@@ -131,3 +139,92 @@ curl \
...
+| `:organization_name` | The name of the organization whose team tokens you want to list. |
+
+This endpoint returns object metadata and does not include secret authentication details of any token. You can only view a token when you create it and cannot recover it later.
+By default, this endpoint returns tokens by ascending expiration date.
+
+| Status  | Response                                                | Reason                                                                                |
+| ------- | ------------------------------------------------------- | ------------------------------------------------------------------------------------- |
+| [200][] | [JSON API document][] (`type: "team-tokens"`)           | The request was successful                                                            |
+| [200][] | Empty [JSON API document][]                             | The specified organization has no team tokens                                        |
+| [404][] | [JSON API error object][]                               | Organization not found                                                                |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/team-tokens.mdx:160]:
```suggestion
| [404][] | [JSON API error object][]                               | Organization not found.                                                                |
```

File: website/docs/cloud-docs/users-teams-organizations/teams.mdx [hunk 12]
```diff
@@ -29,7 +29,7 @@ In addition to the HCP Terraform UI, you can use the following methods to manage
 
 - [Teams API](/terraform/cloud-docs/api-docs/teams) to list, create, update, and delete teams
 - [Team Members API](/terraform/cloud-docs/api-docs/team-members) to add and delete users from teams
-- [Team Tokens API](/terraform/cloud-docs/api-docs/team-tokens) to generate and delete tokens
+- [Team Tokens API](/terraform/cloud-docs/api-docs/team-tokens) to list, generate, and delete tokens
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/teams.mdx:32]:
```suggestion
- [Team Tokens API](/terraform/cloud-docs/api-docs/team-tokens) to generate and delete tokens and list an organization's team tokens
```

**Review by rkoron007 — APPROVED** [review]
Looks great! Added some small suggestions, but after them this looks good to go! Thank you for all your work on this! ✨

File: website/docs/cloud-docs/api-docs/team-tokens.mdx [hunk 13]
```diff
@@ -1,7 +1,7 @@
 ---
 page_title: Team Tokens - API Docs - HCP Terraform
 description: >-
-  Use the `/authentication-token` endpoint to manage team API tokens. Generate and delete team API tokens using the HTTP API.
+  Use the `/authentication-token` endpoint to manage team API tokens. Generate, delete and list team API tokens using the HTTP API.
```
[the remark this answers is not in the document]
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/team-tokens.mdx:4]: Just saw this was `resolved`, but it still looks like this is missing the comma, so I'm reopening it.

File: website/docs/cloud-docs/api-docs/team-tokens.mdx [hunk 14]
```diff
@@ -38,6 +38,8 @@ description: >-
 
 # Team Token API
 
+Team API tokens allow access to the workspaces that the team has access to, without being tied to any specific user. A user can create, delete and list tokens in the organization.
```
[the remark this answers is not in the document]
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/team-tokens.mdx:41]: I saw this was resolved, but I want to push back here because the current version is a bit misleading, hence my suggestion. 

`A user can create, delete and list tokens in the organization.` is correct, but not really what this page is about, so I suggested `You can create and delete team tokens and list an organization's team tokens.` because separating the two thoughts is more true to the API endpoints on this page. If you disagree, please let me know!

File: website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx [hunk 15]
```diff
@@ -21,7 +21,7 @@ API tokens may belong directly to a user. User tokens are the most flexible toke
 
 API tokens may belong to a specific team. Team API tokens allow access to the workspaces that the team has access to, without being tied to any specific user.
 
-To manage the API token for a team, go to **Organization settings > Teams > (desired team)** and use the controls under the "Team API Token" header.
+To manage the API token for a team, go to **Organization settings > API Tokens > Team Token tab** to create a new token or search for a team that you would like to manage the API token for.
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/api-tokens.mdx:24]: For brevity, we can shorten this to:
```suggestion
Navigate to the **Organization settings > API Tokens > Team Token** tab to manage API tokens for a team or create new team tokens.
```

File: website/docs/cloud-docs/users-teams-organizations/teams.mdx (hunk 12 above)
Comment (rkoron007) [reply website/docs/cloud-docs/users-teams-organizations/teams.mdx:32]: Reopening because I think we need to separate `generate` and `delete` from `list` because the scope is different here, let me know if you disagree!

File: website/docs/cloud-docs/users-teams-organizations/teams.mdx [hunk 16]
```diff
@@ -119,7 +119,7 @@ We recommend making most (or all) teams visible to simplify workspace administra
 
 ### API Tokens
 
-Each team can have an API token that is not associated with a specific user. You can manage this API token from the team's settings page under **Team API Token** . Refer to [Team API Tokens](/terraform/cloud-docs/users-teams-organizations/api-tokens#team-api-tokens) for details.
+Each team can have an API token that is not associated with a specific user. You can manage this API token from the API tokens page, in the team token tab, where tokens can be created, regenerated and deleted. Refer to [Team API Tokens](/terraform/cloud-docs/users-teams-organizations/api-tokens#team-api-tokens) for details.
```
Comment (rkoron007) [inline website/docs/cloud-docs/users-teams-organizations/teams.mdx:122]: Suggestions for active voice and specifying where the API tokens page is:
```suggestion
Each team can have an API token that is not associated with a specific user. You can manage a team's API token from the **Organization settings > API Tokens > Team Token** page. On the API token page you can create, regenerate, and delete team tokens. Refer to [Team API Tokens](/terraform/cloud-docs/users-teams-organizations/api-tokens#team-api-tokens) for details.
```

File: website/docs/cloud-docs/api-docs/changelog.mdx [hunk 17]
```diff
@@ -9,6 +9,9 @@ description: >-
 
 Keep track of changes to the API for HCP Terraform and Terraform Enterprise.
 
+## 2024-06-28
+* Add documentation for new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/team-tokens).
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/changelog.mdx:13]:
```suggestion
* Add documentation for a new API endpoint that lists an [organization's team tokens](/terraform/cloud-docs/api-docs/team-tokens).
```

**Review by rkoron007 — APPROVED** [review]
Looking great! ✨ I unresolved some of my suggestions from the last round of review because I do still think they'd improve the writing flow, but let me know if you disagree with any of them!
